music/src/transcription/audio_utils.ts

```diff
@@ -134,26 +132,54 @@ function powerToDb(spec: Float32Array[], amin = 1e-10, topDb = 80.0) {
   return logSpec;
 }
 
-async function resampleAndSetChannels(
-    audioBuffer: AudioBuffer, targetSr = SAMPLE_RATE, numChannels = 1) {
-  if (audioBuffer.sampleRate === targetSr &&
-      audioBuffer.numberOfChannels === numChannels) {
-    return audioBuffer;
+function getMonoAudio(audioBuffer: AudioBuffer) {
+  if (audioBuffer.numberOfChannels === 1) {
+    return audioBuffer.getChannelData(0);
+  }
+  if (audioBuffer.numberOfChannels !== 2) {
+    throw Error(
+        `${audioBuffer.numberOfChannels} channel audio is not supported.`);
+  }
+  const ch0 = audioBuffer.getChannelData(0);
+  const ch1 = audioBuffer.getChannelData(1);
+
+  const mono = new Float32Array(audioBuffer.length);
+  for (let i = 0; i < audioBuffer.length; ++i) {
+    mono[i] = (ch0[i] + ch1[i]) / 2;
+  }
+  return mono;
+}
+
+async function resampleAndMakeMono(
+    audioBuffer: AudioBuffer, targetSr = SAMPLE_RATE) {
+  if (audioBuffer.sampleRate === targetSr) {
+    return getMonoAudio(audioBuffer);
   }
   const sourceSr = audioBuffer.sampleRate;
   const lengthRes = audioBuffer.length * targetSr / sourceSr;
-  // tslint:disable-next-line:no-any
-  const appeaseTsLintWindow = (window as any);
-  const offlineCtx = new (
-      appeaseTsLintWindow.OfflineAudioContext ||
-      appeaseTsLintWindow.webkitOfflineAudioContext)(
-      numChannels, lengthRes, targetSr);
-
-  const bufferSource = offlineCtx.createBufferSource();
-  bufferSource.buffer = audioBuffer;
-  bufferSource.connect(offlineCtx.destination);
-  bufferSource.start();
-  return offlineCtx.startRendering();
+  if (!appeaseTsLintWindow.webkitOfflineAudioContext) {
+    const offlineCtx: OfflineAudioContext =
+        new appeaseTsLintWindow.OfflineAudioContext(1, lengthRes, targetSr);
+
+    const bufferSource = offlineCtx.createBufferSource();
+    bufferSource.buffer = audioBuffer;
+    bufferSource.connect(offlineCtx.destination);
+    bufferSource.start();
+    return offlineCtx.startRendering().then(
+        (buffer: AudioBuffer) => buffer.getChannelData(0));
+  } else {
+    // Safari does not support resampling with WebAudio.
```

Comment:
Can you add a note about this on the main Onset & Frames documentation, please? Maybe a little section about like "Known Issues"?

Reply:
what main documentation? :)

That's also a TODO.